services/frontend-service/src/ui/ReleaseDialog.tsx

```diff
@@ -270,6 +270,34 @@ const DeployButton = (props: {
     }
 };
 
+const UndeployButton = (props: {
+    openDialog?: () => void; //
+    state?: string; //
+    applicationName: string; //
+}) => {
+    const buttonMsg = 'Prepare to Undeploy';
+    const tooltipMsg =
+        'This will create a new version that is empty. Use this only for services that are not needed anymore.';
+    switch (props.state) {
+        case 'waiting':
+            return (
+                <Tooltip title={tooltipMsg}>
+                    <Button variant="contained" onClick={props.openDialog}>
+                        {buttonMsg}
+                    </Button>
+                </Tooltip>
+            );
+        case 'pending':
+            return <div>..waiting...</div>;
+        case 'resolved':
```

Comment:
I don't think this is very ideal. maybe we can display a `disabled` button or just `"undeployment ready"` / `"ready for deletion"` 

also it may be useful to prevent doing this if the latest version is already an undeploy version. => also `disabled` button

Reply:
> also it may be useful to prevent doing this if the latest version is already an undeploy version. => also disabled button

I thought about this, and would like to have it, but it seems rather complicated.
Just because we have multiple envs, some can have undeploy version while others do not. So I'd like to keep this simple for now

Reply:
> I don't think this is very ideal. maybe we can display a disabled button or just "undeployment ready" / "ready for deletion"

Yes you are right, I will change this

Reply:
done